os/arch/arm/src/amebad/amebad_i2c.c

```diff
@@ -0,0 +1,1237 @@
+#include <stdint.h>
+#include <stdbool.h>
+#include <semaphore.h>
+#include <errno.h>
+#include <debug.h>
+
+#include <tinyara/arch.h>
+#include <tinyara/irq.h>
+#include <tinyara/clock.h>
+#include <tinyara/semaphore.h>
+#include <tinyara/i2c.h>
+#include <tinyara/mm/mm.h>
+
+#include "mbed/targets/hal/rtl8721d/PinNames.h"
+
+#include <arch/board/board.h>
+
+#include "up_arch.h"
+
+#include "amebad_i2c.h"
+
+/************************************************************************************
+ * Pre-processor Definitions
+ ************************************************************************************/
+
+/* Configuration ********************************************************************/
+/* CONFIG_I2C_POLLED may be set so that I2C interrupts will not be used.  Instead,
+ * CPU-intensive polling will be used.
+ */
+
+/* Interrupt wait timeout in seconds and milliseconds */
+
+#if !defined(CONFIG_AMEBAD_I2CTIMEOSEC) && !defined(CONFIG_AMEBAD_I2CTIMEOMS)
+#define CONFIG_AMEBAD_I2CTIMEOSEC 0
+#define CONFIG_AMEBAD_I2CTIMEOMS  500	/* Default is 500 milliseconds */
+#elif !defined(CONFIG_AMEBAD_I2CTIMEOSEC)
+#define CONFIG_AMEBAD_I2CTIMEOSEC 0	/* User provided milliseconds */
+#elif !defined(CONFIG_AMEBAD_I2CTIMEOMS)
+#define CONFIG_AMEBAD_I2CTIMEOMS  0	/* User provided seconds */
+#endif
+
+/* Interrupt wait time timeout in system timer ticks */
+
+#ifndef CONFIG_AMEBAD_I2CTIMEOTICKS
+#define CONFIG_AMEBAD_I2CTIMEOTICKS \
+	(SEC2TICK(CONFIG_AMEBAD_I2CTIMEOSEC) + MSEC2TICK(CONFIG_AMEBAD_I2CTIMEOMS))
+#endif
+
+#ifndef CONFIG_AMEBAD_I2C_DYNTIMEO_STARTSTOP
+#define CONFIG_AMEBAD_I2C_DYNTIMEO_STARTSTOP TICK2USEC(CONFIG_AMEBAD_I2CTIMEOTICKS)
+#endif
+
+/* Debug ****************************************************************************/
+
+/* I2C event trace logic.  NOTE:  trace uses the internal, non-standard, low-level
+ * debug interface syslog() but does not require that any other debug
+ * is enabled.
+ */
+
+#ifndef CONFIG_I2C_TRACE
+#define amebad_i2c_tracereset(p)
+#define amebad_i2c_tracenew(p, s)
+#define amebad_i2c_traceevent(p, e, a)
+#define amebad_i2c_tracedump(p)
+#endif
+
+#ifndef CONFIG_I2C_NTRACE
+#define CONFIG_I2C_NTRACE 32
+#endif
+
+#define AMEBAD_I2C_MASTER    1
+#define AMEBAD_I2C_SLAVE     2
+
+#define I2C_MASTER_DEVICE		1
+#define I2C_SLAVE_DEVICE		2
+
+#define MKI2C_OUTPUT(p) (((p) & GPIO_PADMUX_MASK) | \
+			 IOMUX_OPENDRAIN | IOMUX_DRIVE_33OHM | \
+			 IOMUX_SLEW_SLOW | (5 << GPIO_ALT_SHIFT) | \
+			 IOMUX_PULL_NONE | GPIO_OUTPUT_ONE)
+
+#define MKI2C_INPUT(p) (((p) & GPIO_PADMUX_MASK) | \
+			IOMUX_DRIVE_HIZ | IOMUX_SLEW_SLOW | \
+			IOMUX_CMOS_INPUT | (5 << GPIO_ALT_SHIFT) | \
+			IOMUX_PULL_NONE)
+
+#undef i2cerr
+#undef i2cinfo
+#if defined(DEBUG_IMX_I2C)
+#define i2cinfo(format, ...)   printf(format, ##__VA_ARGS__)
+#define i2cerr(format, ...)   printf(format, ##__VA_ARGS__)
+#else
+#define i2cinfo(format, ...)
+//#define nerr(format, ...)
+#endif
+#define i2cerr(format, ...)   printf(format, ##__VA_ARGS__)
```

Comment:
Could you remove them? They are already in debug.h.

Reply:
Removed.